src/main/java/org/assertj/core/api/AbstractIterableAssert.java

```diff
@@ -1515,4 +1517,10 @@ public S withFailMessage(String newErrorMessage, Object... args) {
   public S withThreadDumpOnError() {
     return super.withThreadDumpOnError();
   }
+  
+  @SuppressWarnings({ "rawtypes", "unchecked" })
+  public IterableSizeAssert<T, Integer> size() {
```

Comment:
try to declare returning an `AbstractIterableAssert`, I don't see a need to return an `IterableSizeAssert` (but keep building an `IterableSizeAssert`) and see if it solved the cast pb...